File: 0761-special-binary-string/0761-special-binary-string.cpp
class Solution {
public:
    string makeLargestSpecial(string s) {
        int count = 0, i = 0;
        vector<string> res;
        
        for (int j = 0; j < s.length(); j++) {
            count += (s[j] == '1') ? 1 : -1;
            if (count == 0) {
                res.push_back('1' + makeLargestSpecial(s.substr(i + 1, j - i - 1)) + '0');
                i = j + 1; 
            }
        }
        sort(res.begin(), res.end(), greater<string>());
        string result;
        for (const string& str : res) {
            result += str;
        }
        return result;
    }
};

File: 0761-special-binary-string/0761-special-binary-string.py
class Solution:
    def makeLargestSpecial(self, s: str) -> str:
        count = 0
        i = 0
        res = []
        for j in range(len(s)):
            count += 1 if s[j] == '1' else -1
            if count == 0:
                res.append('1' + self.makeLargestSpecial(s[i + 1:j]) + '0')
                i = j + 1 
        res.sort(reverse=True)
        return ''.join(res)